d3a_interface/data_classes.py

```diff
@@ -131,6 +132,23 @@ def serializable_dict(self) -> Dict:
                 "seller_id": self.seller_id,
                 }
 
+    @staticmethod
+    def from_dict(offer: Dict) -> "Offer":
+        """Deserialize an offer dict."""
+
+        return Offer(
+            id=offer["id"],
+            time=str_to_pendulum_datetime(offer["time"]),
+            energy=offer["energy"],
+            price=offer["energy"] * offer["energy_rate"],
+            original_price=offer.get("original_price"),
+            seller=offer.get("seller"),
+            seller_origin=offer.get("seller_origin"),
+            seller_origin_id=offer.get("seller_origin_id"),
+            seller_id=offer.get("seller_id"),
+            attributes=offer.get("attributes"),
+            requirements=offer.get("requirements"))
```

Comment:
In case the `offer` dictionary isn't correct, this would raise a `KeyError` exception.
I'd suggest catching possible `KeyError`s and raise a new exception (e.g. `OfferDeserializationError`) with a more informative message.
Also, I would really encourage you to modify the `str_to_pendulum_datetime` function to raise a specific exception (it's currently raising the bare `Exception`). This would allow us to also catch possible errors during the datetime conversion and handle them properly.

Reply:
The same comment applies to the `Bid.from_dict()` below.